File: src/main/java/com/trading/kalyani/KTManager/entity/MiniDelta.java
package com.trading.kalyani.KTManager.entity;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.boot.context.properties.bind.DefaultValue;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@Table(name="MiniDelta")
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class MiniDelta {
    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    private Long id;

    private Integer appJobConfigNum;

    private LocalDateTime deltaInstant;

    private String strikePrice;

    private Double rateOI,strikePCR,callOI,putOI,callOIChange,putOIChange;

    @ManyToOne
    @JoinColumn(name = "jobIterationId", referencedColumnName = "id", nullable = false)
    public JobIterationDetails jobIterationDetails;

    @Transient
    private String rateOIColor,strikePCRColor,callOIColor,putOIColor,callOIChangeColor,putOIChangeColor;

}

File: kpn-ui/src/app/models/minidelta.ts
export class MiniDelta {
    id: number;
    deltaInstant: string;
    strikePrice: string;
    rateOI: number;
    strikePCR: number;
    callOI: number;
    putOI: number;
    callOIChange: number;
    putOIChange: number;
    callOIColor: string;
    putOIColor: string;
    callOIChangeColor: string;
    putOIChangeColor: string;
    rateOIColor: string;
    strikePCRColor: string;
    appJobConfigNum?: number; // Optional property to link with AppJobConfig


    constructor() {

        this.id = 0;
        this.deltaInstant = '';
        this.strikePrice ='';
        this.callOI =0.0;
        this.putOI = 0.0;
        this.rateOI = 0.0;
        this.strikePCR = 0.0;
        this.callOIChange = 0.0;
        this.putOIChange = 0.0;
        this.callOIColor = '';
        this.putOIColor = '';
        this.callOIChangeColor = '';
        this.putOIChangeColor = '';
        this.rateOIColor = '';
        this.strikePCRColor = '';
        this.appJobConfigNum = 0;
    }
}
